### ISSUE TITLE
Suggest consider refactoring to minimise Discovery Load and dependency

### ISSUE THREAD

**Issue by pscott-au** [sdondley/moo-google#20]
The current implementation does a full discovery when a dynamic class is instantiated which is a little rude really and certainly not normal practice. I've noticed that there are intermittent declines from Google when I'm accessing this frequently perhaps hinting that this isn't something they encourage. There really should be some kind of memoisation to cache the service definitions as though Google does change their API's I think that checking every call is a little bit of an overkill.

**Comment by sdondley** [comment]
Probably would be good to store it in a cache, maybe even with Storable, and only refresh it if there is an unknown/failed api call.

**Comment by pscott-au** [comment]
yes - and this ties into the persistence classes I would expect depending
on choice of json/DBI/Mongo if they are to be implemented.

**Comment by pscott-au** [comment]
CHI worked nicely for me.
